datasets/HMRC_Trade/main.py

```diff
@@ -6,338 +6,132 @@
-#    Unregistered businesses (those not registered for VAT or Economic Operator Registration and Identification (EORI)).
-#    Due to these experimental statistics being subject to active disclosure controls the data has been suppressed 
-#    according to GSS guidance on disclosure control.  Suppressed cells are shown with an 'S'
-#    """
-#scraper.dataset.comment = 'Supporting tables for the UK trade in goods by business characteristics 2018'
-#scraper.dataset.description = comDesc
-#scraper.dataset.title = 'HMRC Trade in Goods'
-
-with open(out / (fn1 + '.csv-metadata.trig'), 'wb') as metadata:metadata.write(scraper.generate_trig())
-csvw = CSVWMetadata('https://gss-cogs.github.io/family-trade/reference/')
-csvw.create(out / (fn1 + '.csv'), out / ((fn1 + '.csv') + '-schema.json'))
-# -
-
-
-
-
+    employee_count = cell.shift(9,0).fill(DOWN).is_not_blank().is_not_whitespace()
+    trace.Employee_Count("Defined form cell ref J1 down")
+
+    dimensions = [
+        HDim(flow, 'Flow', DIRECTLY, LEFT),
+        HDim(year, 'Period', DIRECTLY, LEFT),
+        HDim(country, 'Country', DIRECTLY, LEFT),
+        HDim(zone, 'Zone', DIRECTLY, LEFT),
+        HDim(business_size, 'Business Size', DIRECTLY, LEFT),
+        HDim(age, 'Age', DIRECTLY, LEFT),
+        HDim(industry_group, 'Industry Group', DIRECTLY, LEFT),
+        HDim(business_count, 'Business Count', DIRECTLY, RIGHT),
+        HDim(employee_count, 'Employee Count', DIRECTLY, RIGHT),
+    ]
+    tidy_sheet = ConversionSegment(tab, dimensions, observations)
+    savepreviewhtml(tidy_sheet, fname = tab.name+ "Preview.html")
+    trace.with_preview(tidy_sheet)
+    trace.store("combined_dataframe", tidy_sheet.topandas())
+df = trace.combine_and_trace(datasetTitle, "combined_dataframe")
+df
+
+df.rename(columns= {'OBS':'Value', 'Period':'Year', 'Flow':'Flow Directions', 'DATAMARKER':'Marker'}, inplace = True)
+
+df['Flow Directions'] = df['Flow Directions'].apply(pathify)
+trace.Flow_Directions("Renamed Flow to Flow Directions")
+
+df['Country'] = df['Country'].apply(pathify)
+trace.Country("Pathified Country")
+
+df['Zone'] =df['Zone'].apply(pathify)
+trace.Zone("Pathified Zone")
+
+df['Business Size'] = df['Business Size'].apply(pathify)
+trace.Business_Size("Pathified Business_Size")
+
+df['Age'] = df['Age'].apply(pathify)
+trace.Age("Pathified Age")
+
+df['Industry Group'] = df['Industry Group'].apply(pathify)
+trace.Industry_Group("Pathified Industry Group")
+
+
+def left(s,amount):
+    return s[:amount]
+def right(s,amount):
+    return s[-amount:]
+def date_time(date):
+    if len(date) == 5:
+        return 'year/' + left(date, 4)
+df['Year'] = df['Year'].astype(str).replace('\.', '', regex=True)
+df['Year'] = df['Year'].apply(date_time)
+trace.Year("Formating to be year/2019")
+
+with pd.option_context('float_format', '{:f}'.format):
+    print(df)
+
+cubes.add_cube(scraper, df.drop_duplicates(), "uk-trade-in-goods-by-business-characteristics-2019")
```

Comment:
change to pass the distribution title also e.g :
cubes.add_cube(scraper, tidy.drop_duplicates(), distribution.title )

Reply:
distribution title is passed to cubes.add_cube() function